### PR TITLE
Adapt maximum brush size depending on available magnifications

### LINKED ISSUE #6060: Maximum brush size should depend on available magnifications
## Detailed Description
Currently, the maximum brush size is always 300. However, if mag 1 does not exist, a higher size should be available.

Potential problems:
- for anisotropic mags (e.g., 2-2-1), tracing in XY affects less voxels than tracing in YZ. how would one switch dynamically between the different limits?
- switching between multiple layers needs to have the side effect of changing the brush size if necessary (can be annoying when peeking into one layer and afterwards having to adjust the size again)

### PR DESCRIPTION
**Pull request by philippotto**
### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz

### Steps to test:
- Create a new annotation using the "..." and restrict the available magnifications (e.g., beginning from mag 4)
- adjust the brush size (the old limit was 300; the new limit should be higher; depending on the existing mags)
- add a second volume layer via the left sidebar
- when switching to the second volume layer, the brush size should clamp again to 300 (if mag 1 exists)
- open a dataset in viewmode to ensure that this also works

### Issues:
- fixes #6060

------
(Please delete unneeded items, merge only when none are left open)
- [x] Updated [(unreleased) changelog](../blob/master/CHANGELOG.unreleased.md#unreleased)
- [X] Ready for review

### CODE REVIEW

**Review by hotzenklotz — APPROVED** [review]
LGTM. I must admit I am not expert on the code-side of things anymore, but changes sound reasonable. 

Testing worked without major any hiccups. One thing I noticed: 
- When switching between two volume layers (one restricted to mag4 and one restricted to mag1), the max brush size is rightfully reduced to 300 when on the mag1 layer. When going back to the mag4 layer the brush size is still at 300 even though more is possible.
- In other words, brush size is global settings and not per layer. It is not a deal break and shouldn't prevent this PR from merging. ideally new layer should be restricted to the same mags anyway in the future, right?

File: frontend/javascripts/oxalis/model/sagas/volumetracing_saga.js [hunk 1]
```diff
@@ -1013,6 +1015,32 @@ function* maintainVolumeTransactionEnds(): Saga<void> {
   yield _takeEvery("FINISH_ANNOTATION_STROKE", markVolumeTransactionEnd);
 }
 
+function* ensureValidBrushSize(): Saga<void> {
+  // A valid brush size needs to be ensured in certain events,
+  // since the maximum brush size depends on the available magnifications
+  // of the active volume layer.
+  // Currently, these events are:
+  // - when webKnossos is loaded
+  // - when a layer's visibility is toggled
```
[the remark this answers is not in the document]
Comment (hotzenklotz) [reply frontend/javascripts/oxalis/model/sagas/volumetracing_saga.js:1024]: Never mind. I understand why now. Testing made it clear when switching between two volume annotation layers :-)

**Comment by hotzenklotz** [comment]
To be honest I would prefer an even bigger brush... 2x is not for my dataset :-P However, the performance/lag already is not great, so maybe I am asking for too much.

 See tiny brush (maxed out) in screenshot:

image

**Comment by philippotto** [comment]
> In other words, brush size is global settings and not per layer. It is not a deal break and shouldn't prevent this PR from merging. ideally new layer should be restricted to the same mags anyway in the future, right?

Yeah, that's a bit unfortunate. However, storing the property per layer could also be annoying, depending on what the user does. For example, if the user annotates stuff of similar sizes in two layers, it could be annoying if the brush size resets to an old value when switching to another layer. I'd keep it as is, because it's simpler and we should gather more realistic feedback on that.

> To be honest I would prefer an even bigger brush... 2x is not for my dataset :-P However, the performance/lag already is not great, so maybe I am asking for too much.
> 
> See tiny brush (maxed out) in screenshot:

If you are annotating such large areas, couldn't you restrict the available magnifications even further? If it's necessary to have broad brush strokes AND fine regions, it might make sense to use two different layers for that? As you say, the performance restricts the size effectively and I don't expect any easy wins, unfortunately. The trace tool might be a bit helpful still..
